libvirt/uri/ssh.go

```diff
@@ -53,17 +74,27 @@ func (u *ConnectionURI) parseAuthMethods() []ssh.AuthMethod {
 			agentClient := agent.NewClient(conn)
 			result = append(result, ssh.PublicKeysCallback(agentClient.Signers))
 		case "privkey":
-			sshKey, err := os.ReadFile(os.ExpandEnv(sshKeyPath))
-			if err != nil {
-				log.Printf("[ERROR] Failed to read ssh key: %v", err)
-				continue
-			}
+			for _,keypath := range sshKeyPaths {
+				log.Printf("[DEBUG] Reading ssh key '%s'", keypath)
+				path := os.ExpandEnv(keypath)
+				if strings.HasPrefix(path, "~/") {
+					home, err := os.UserHomeDir()
+					if err == nil {
+						path = strings.Replace(path, "~", home, 1)
```

Comment:
Instead of replacing, you could just join:

```go
if strings.HasPrefix(path, "~/") {
    dirname, _ := os.UserHomeDir()
    path = filepath.Join(dirname, path[2:])	
}
```